pyfritzhome/devicetypes/fritzhomedevicepowermeter.py

```diff
@@ -36,10 +36,18 @@ def _update_powermeter_from_node(self, node):
         self.energy = int(val.findtext("energy"))
         try:
             self.voltage = int(val.findtext("voltage"))
-            self.current = self.power / self.voltage * 1000
         except Exception:
             pass
 
+        if (
```

Comment:
What about catching the exceptions;

```
try:
  self.current = self.power / self.voltage * 1000
except ZeroDivisionError:
  self.current = None
except TypeError:
  self.current = None
```

Reply:
I always prefer to handle possible situations (_as far as possible_), over try-catch constructs.
Sure, if there it is function call from another library, then try-catch is the most coinvent way, but in this case, the properties and their possible states/values are well known.
But at the end, it is your decision - if you want, I will rewrite it as a try-catch like:
```python
try:
  self.current = self.power / self.voltage * 1000
except Exception:
  self.current = None
```